### PR TITLE
8332849: Update doc/testing.{md,html} (spelling and stale information)

### PR DESCRIPTION
**Pull request by vidmik**
Update the testing doc to remove some stale information (AOT_MODULES, removed in [JDK-8264805](https://bugs.openjdk.org/browse/JDK-8264805)) and fix some spelling issues.

Testing: tier1

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8332849](https://bugs.openjdk.org/browse/JDK-8332849): Update doc/testing.{md,html} (spelling and stale information) (**Enhancement** - P4)

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to [859e52d1](https://git.openjdk.org/jdk/pull/19375/files/859e52d17957ec3cfe7b314d5cf3c178c7f9b5a8)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19375/head:pull/19375` \
`$ git checkout pull/19375`

Update a local copy of the PR: \
`$ git checkout pull/19375` \
`$ git pull https://git.openjdk.org/jdk.git pull/19375/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19375`

View PR using the GUI difftool: \
`$ git pr show -t 19375`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19375.diff">https://git.openjdk.org/jdk/pull/19375.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19375#issuecomment-2128188457)

### CODE REVIEW

**Review by irisclark — APPROVED** [review]

**Review by magicus — APPROVED** [review]
Thanks! ❤️

File: doc/testing.html [hunk 1]
```diff
@@ -363,12 +363,12 @@ <h2 id="test-suite-control">Test suite control</h2>
 concurrency level to 1 and the timeout factor to 8. This is equivalent
 to setting <code>JTREG_JOBS=1 JTREG_TIMEOUT_FACTOR=8</code>, but using
 the keyword format means that the <code>JTREG</code> variable is parsed
-and verified for correctness, so <code>JTREG="TMIEOUT_FACTOR=8"</code>
-would give an error, while <code>JTREG_TMIEOUT_FACTOR=8</code> would
+and verified for correctness, so <code>JTREG="TIMEOUT_FACTOR=8"</code>
```
Comment (djelinski) [inline doc/testing.html:366]: this was a deliberate typo

Comment (erikj79) [reply doc/testing.html:366]: Indeed, these two were intentional to demonstrate the error handling of the JTREG make variable.

**Review by erikj79 — CHANGES_REQUESTED** [review]

File: doc/testing.html (hunk 1 above)
Comment (vidmik) [reply doc/testing.html]: Well that's embarrassing.. Thank you for the catch!

Fixed (reverted).

**Review by erikj79 — APPROVED** [review]

**Review by djelinski — APPROVED** [review]
LGTM

**Review by irisclark — APPROVED** [review]

**Comment by vidmik** [comment]
Thank you for all the reviews!

/integrate

**Pull request closed without merging**